### ISSUE TITLE
wineWowPackages do not build on darwin x86_64

### ISSUE THREAD

**Issue by maisem** [NixOS/nixpkgs#137734]
### Describe the bug
`wineWowPackages.stable` doesn't build on x86_64-darwin
```
❯ nix-env -iA nixpkgs.wineWowPackages.stable
installing 'wine-wow-6.0'
error: Multilib clang-7.1.0 not supported for 'x86_64-darwin'
(use '--show-trace' to show detailed location information)
```

### Steps To Reproduce
1. on darwin with nixpkgs run
   ```nix-env -iA nixpkgs.wineWowPackages.stable```

### Expected behavior
Expected wine to install properly

### Additional context
Problem looks to be stemming from 
```
map (x: "${lib.getLib x}/lib") ([ stdenv.cc.cc ] ++ buildInputs)
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/emulators/wine/base.nix#L92

### Notify maintainers
@avnik @raskin @bendlas

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.3.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4pre20210326_dd77f71`
 - channels(maisem): `"darwin, home-manager-20.09, nixpkgs-21.11pre294471.51bb9f3e9ab"`
 - channels(root): `"darwin, nixpkgs-21.05pre278688.c0e88185200"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by bendlas** [comment]
WineWOW needs a multilib compiler, because it compiles both 32 and 64 bit code in a single build. If nix on darwin doesn't provide that, that would be the place to start.

Unfortunately, I don't have a mac and know little about how nix works on it.

Do wine32 and wine64 both work?

**Comment by reckenrode** [comment]
Since the 10.14 SDK, it’s not possible to link 32-bit applications on Darwin. While x86_64-darwin is currently using the 10.12 SDK, I don’t know whether it can link 32-bit applications either. The future will be using Wine’s new WoW64 support. However, from what I understand, it doesn’t yet support Darwin.

**Comment by Atemu** [comment]
What's the status on this?

IIRC Wine upstream should support WoW on Darwin now.

**Comment by reckenrode** [comment]
I’ve read the conversion is not quite done yet, but it wouldn’t hurt to try building it anyway. The previous failures were during the build, so any problems should be obvious right away. I can look at it sometime this week.

**Comment by reckenrode** [comment]
@Gcenx pinged me regarding a set of out of tree patches to make this work on Darwin. There are some caveats (e.g., printing is disabled), but it would allow some 32-bit Windows applications to run using Wine from nixpkgs.

**Comment by reckenrode** [comment]
The WoW64 packages now build on Darwin. Is this still an issue?

**Comment by Atemu** [comment]
The title of this issue is that they did not build. If they build now then this issue is resolved.

**Comment by bendlas** [comment]
> The title of this issue is that they did not build. If they build now then this issue is resolved.

WoW64 != WoW

Though I think WoW64 is the future of wine, so we might do without WoW

**Comment by reckenrode** [comment]
> The title of this issue is that they did not build. If they build now then this issue is resolved.

The problem is WoW builds 32-bit libraries, which is effectively impossible in nixpkgs with our tooling. After 24.11 is released, the minimum version will be updated to 11.0, which does not support 32-bit at an OS level. Even if the WoW packages could be built, they wouldn’t work.

Wine has to take special steps to run 32-bit code on macOS 10.15 and newer. The supported way to do that is using WoW64. CrossOver used to use a different way that required a custom clang, but I don’t think they use that anymore.

Given the above, the question is whether WoW64 is an acceptable solution. If not, then this should be closed wontfix because it’ll never be implemented in nixpkgs for Darwin.

Once https://github.com/NixOS/nixpkgs/pull/346043 is merged, I’ll be updating Wine to use the 14.4 SDK to take advantage of an API (when available) added in 14.4 for improved futex support. The 14.4 SDK definitely does not support building for 32-bit Darwin. It won’t even eval for it.
